DatadogLogs/Sources/Logs.swift

```diff
@@ -0,0 +1,110 @@
+        /// By default sampling is disabled, meaning that all logs are being processed).
+        public var sampleRate: Float
+
+        /// Overrides the custom server endpoint where Logs are sent.
+        public var customEndpoint: URL?
+
+        /// Overrides the main bundle instance.
+        public var bundle: Bundle = .main
+
+        /// Overrides the current process info.
+        internal var processInfo: ProcessInfo = .processInfo
+
+        /// Sets the custom mapper for `LogEvent`. This can be used to modify logs before they are send to Datadog.
+        ///
+        /// - Parameter mapper: the closure taking `LogEvent` as input and expecting `LogEvent` as output.
+        /// The implementation should obtain a mutable version of the `LogEvent`, modify it and return it. Returning `nil` will result
+        /// with dropping the Log event entirely, so it won't be send to Datadog.
+        public mutating func eventMapper(_ mapper: @escaping EventMapperClosure) {
+            self.mapper = SyncLogEventMapper(mapper)
+        }
+
+        /// Sets the custom mapper for `LogEvent`. This can be used to modify logs before they are send to Datadog.
+        ///
+        /// The implementation should obtain a mutable version of the `LogEvent`, modify it and return it. Returning `nil` will result
+        /// with dropping the Log event entirely, so it won't be send to Datadog.
+        internal var mapper: LogEventMapper?
+
+        /// Creates a Logs configuration object.
+        ///
+        /// - Parameters:
+        ///   - eventMapper: The custom mapper for `LogEvent`. This can be used to modify logs before they are send to Datadog.
+        ///   - sampleRate: The sample rate for logging.
+        ///   - sampleRate: Overrides the custom server endpoint where Logs are sent.
+        ///   - bundle: Overrides the main bundle instance.
+        public init(
+            eventMapper: EventMapperClosure? = nil,
+            sampleRate: Float = 100,
+            customEndpoint: URL? = nil,
+            bundle: Bundle = .main
+        ) {
+            self.mapper = eventMapper.map(SyncLogEventMapper.init)
+            self.sampleRate = sampleRate
+            self.customEndpoint = customEndpoint
+            self.bundle = bundle
+        }
+    }
+
+    /// Enables the Datadog Logs feature.
+    ///
+    /// - Parameters:
+    ///   - configuration: The Logs configuration.
+    ///   - core: The instance of Datadog SDK to enable Logs in (global instance by default).
+    public static func enable(
+        with configuration: Configuration = .init(),
+        in core: DatadogCoreProtocol = CoreRegistry.default
+    ) {
+        let applicationBundleIdentifier = configuration.bundle.bundleIdentifier ?? "unknown"
+        let debug = configuration.processInfo.arguments.contains(LaunchArguments.Debug)
+        let feature = LogsFeature(
+            logEventMapper: configuration.mapper,
+            dateProvider: SystemDateProvider(),
```

Comment:
suggestion/ If we move `dateProvider` to `Logs.Configuration`:
```swift
struct Logs.Configuration {
   internal var dateProvider: DateProvider = SystemDateProvider()
}
```
then we will no longer need to mock `LogsFeature` in tests and register it to `core` every time. Both will be equal:
```swift
Logs.enable(with: .init(dateProvider: DateProviderMock()), in: core)
```
```swift
let logs: LogsFeature = .mockWith(dateProvider: DateProviderMock())
core.register(logs)
```
This way we can reduce the number of feature mocks.